### PR TITLE
Python: Refactoring and fix bug. AuthorRole instead of string literals.

### PR DESCRIPTION
**Pull request by stefan521**
### Motivation and Context

There's an enum called `AuthorRole`. I replaced string literals with the enum values. PyCharm highlights this in some cases: Expected type 'AuthorRole', got 'str' instead

I also fixed a bug in _function_invocation_filters_stream.py_ where the argument name was wrong. `author="assistant"` instead of `role="assistant"`

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by moonbox3 — APPROVED** [review]
LGTM

**Comment by moonbox3** [comment]
Hi @stefan521, could you resolve the one file conflict, please? Thanks!

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/connectors/ai/ollama/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py">ollama_chat_completion.py</a></td><td>67</td><td>6</td><td>91%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py#L56">56</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py#L94">94</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py#L116">116</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py#L133">133</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py#L167">167</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py#L188">188</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py">open_ai_chat_completion_base.py</a></td><td>215</td><td>94</td><td>56%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L97">97</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L117">117</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L142-L146">142&ndash;146</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L170">170</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L174">174</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L190-L195">190&ndash;195</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L212-L240">212&ndash;240</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L243-L254">243&ndash;254</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L269-L276">269&ndash;276</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L287-L295">287&ndash;295</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L311-L318">311&ndash;318</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L339">339</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L347">347</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L353-L356">353&ndash;356</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L368-L371">368&ndash;371</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L402">402</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L441">441</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L446">446</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L451-L456">451&ndash;456</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L460-L467">460&ndash;467</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L471-L483">471&ndash;483</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L505">505</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L515-L526">515&ndash;526</a></td></tr><tr><td colspan="5"><b>semantic_kernel/contents</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py">function_result_content.py</a></td><td>57</td><td>12</td><td>79%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L50">50</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L55">55</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L66">66</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L81">81</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L93">93</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L107">107</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L112">112</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0b29101304a52bdf759f901484e99b3e4f486277/python/semantic_kernel/contents/function_result_content.py#L119-L123">119&ndash;123</a></td></tr><tr><td><b>TOTAL</b></td><td><b>6095</b></td><td><b>785</b></td><td><b>87%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1402 | 1 :zzz: | 0 :x: | 0 :fire: | 23.046s :stopwatch: |

**Comment by stefan521** [comment]
Hi @moonbox3 , thanks for reviewing! I've solved the conflicts.

File: python/tests/unit/contents/test_chat_message_content.py [hunk 1]
```diff
@@ -12,20 +12,20 @@
...
-    message = ChatMessageContent(role="user", content="Hello, world!")
+    message = ChatMessageContent(role=AuthorRole.USER, content="Hello, world!")
     assert message.role == AuthorRole.USER
     assert message.content == "Hello, world!"
     assert len(message.items) == 1
 
 
 def test_cmc_str():
-    message = ChatMessageContent(role="user", content="Hello, world!")
+    message = ChatMessageContent(role=AuthorRole.USER, content="Hello, world!")
```
Comment (eavanvalkenburg) [inline python/tests/unit/contents/test_chat_message_content.py:22]: Please keep this as a string and add a assert for the test, the return value from a LLM is a str as well!

File: python/tests/unit/contents/test_streaming_chat_message_content.py [hunk 2]
```diff
@@ -15,21 +15,21 @@
...
-    message = StreamingChatMessageContent(choice_index=0, role="user", content="Hello, world!")
+    message = StreamingChatMessageContent(choice_index=0, role=AuthorRole.USER, content="Hello, world!")
     assert message.role == AuthorRole.USER
     assert message.content == "Hello, world!"
     assert len(message.items) == 1
 
 
 def test_scmc_str():
-    message = StreamingChatMessageContent(choice_index=0, role="user", content="Hello, world!")
+    message = StreamingChatMessageContent(choice_index=0, role=AuthorRole.USER, content="Hello, world!")
```
Comment (eavanvalkenburg) [inline python/tests/unit/contents/test_streaming_chat_message_content.py:25]: Same as for CMC

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]

**Comment by stefan521** [comment]
@eavanvalkenburg got it! I’ll update the PR tonight. Thanks for reviewing!

**Pull request merged**